pkg/policyconditions/conditions.go

```diff
@@ -178,75 +177,9 @@ func update(apiWriter client.Client, apiReader client.Reader, policyReader clien
 		if err != nil {
 			return errors.Wrap(err, "getting nodes running kubernets-nmstate pods failed")
 		}
-		numberOfNmstateMatchingNodes := len(nmstateMatchingNodes)
-		numberOfReadyNmstateMatchingNodes := len(node.FilterReady(nmstateMatchingNodes))
-		numberOfNotReadyNmstateMatchingNodes := numberOfNmstateMatchingNodes - numberOfReadyNmstateMatchingNodes
-
-		// Let's get conditions with true status count filtered by policy generation
-		enactmentsCountByCondition := enactmentconditions.Count(enactments, policy.Generation)
-
-		numberOfFinishedEnactments := enactmentsCountByCondition.Available() +
-			enactmentsCountByCondition.Failed() +
-			enactmentsCountByCondition.Aborted()
-
-		logger.Info(
-			fmt.Sprintf("numberOfNmstateMatchingNodes: %d, enactments count: %s",
-				numberOfNmstateMatchingNodes,
-				enactmentsCountByCondition),
-		)
+		setPolicyStatus(policy, &nmstateMatchingNodes, &enactments, &logger)
```

Comment:
I think that abstracting the status logic to a separate function could be a benefit, but can we limit the function to do only that?
And collect/calculate the node statuses outside of this function